raiden/tests/unit/test_cli.py

```diff
@@ -32,6 +43,50 @@ def test_cli_version(cli_runner):
     assert result_expected_keys == result_json.keys()
     assert result.exit_code == 0
 
+    result = cli_runner(run, ["--version"])
+    assert "Hint: Use '" in result.output and "version' instead" in result.output
+    assert result.exit_code == 0
+
+
+def mock_raises(exception):
+    def f(*_, **__):
+        raise exception
+
+    return f
+
+
+def test_run_error_reporting(cli_runner, monkeypatch):
+    result = cli_runner(run, ["--transport", "udp"])
+    assert "Invalid value" in result.output and "--transport" in result.output
+    assert result.exit_code != 0
+
+    caught_exceptions = {
+        APIServerPortInUseError(): ReturnCode.PORT_ALREADY_IN_USE,
+        ConfigurationError(): ReturnCode.CONFIGURATION_ERROR,
+        ConnectTimeout(): ReturnCode.GENERIC_COMMUNICATION_ERROR,
+        ConnectionError(): ReturnCode.GENERIC_COMMUNICATION_ERROR,
+        EthereumNonceTooLow(): ReturnCode.ETH_ACCOUNT_ERROR,
+        EthNodeInterfaceError(): ReturnCode.ETH_INTERFACE_ERROR,
+        KeystoreAuthenticationError(): ReturnCode.ETH_ACCOUNT_ERROR,
+        KeystoreFileNotFound(): ReturnCode.ETH_ACCOUNT_ERROR,
+        RaidenUnrecoverableError(): ReturnCode.FATAL,
+        ReplacementTransactionUnderpriced(): ReturnCode.ETH_ACCOUNT_ERROR,
+        RequestsConnectionError(): ReturnCode.GENERIC_COMMUNICATION_ERROR,
+        Exception(): ReturnCode.FATAL,
+    }
+
+    for exception, code in caught_exceptions.items():
+        monkeypatch.setattr(MatrixRunner, "run", mock_raises(exception))
+        result = cli_runner(run, "--accept-disclaimer")
+        assert result.exception.code == code
+
+
+def test_check_is_supported_unknown_client():
+    supported, client, version = is_supported_client("Aleth//v1.2.1")
```

Comment:
this is already tested `test_cli.py`

Reply:
Sure. This is the test we already had, I just moved it to group similiar tests together.